### PR TITLE
Prefer forkchoice head as latest block in RPC

### PR DESCRIPTION
**Pull request by yperbasis**
See [Erigon Hive test failure](https://notes.ethereum.org/@3a9tFrpLTxq1U828RfGEEw/BJoZYjPMc#Latest-Block-after-NewPayload).

### CODE REVIEW

File: turbo/stages/headerdownload/header_algos.go [hunk 1]
```diff
@@ -1289,24 +1289,24 @@ func (hd *HeaderDownload) ClearPendingPayloadStatus() {
 	hd.pendingPayloadStatus = common.Hash{}
 }
 
-func (hd *HeaderDownload) GetPendingHeader() (common.Hash, uint64) {
+func (hd *HeaderDownload) GetUnsettledForchoice() (*engineapi.ForkChoiceMessage, uint64) {
```
Comment (enriavil1) [inline turbo/stages/headerdownload/header_algos.go:1292]: `GetUnsettledForchoice` has a typo

File: turbo/stages/headerdownload/header_algos.go [hunk 2]
```diff
@@ -1289,24 +1289,24 @@ func (hd *HeaderDownload) ClearPendingPayloadStatus() {
...
 	hd.lock.RLock()
 	defer hd.lock.RUnlock()
-	return hd.pendingHeaderHash, hd.pendingHeaderHeight
+	return hd.unsettledForkChoice, hd.unsettledHeadHeight
 }
 
-func (hd *HeaderDownload) SetPendingHeader(blockHash common.Hash, blockHeight uint64) {
+func (hd *HeaderDownload) SetUnsettledForchoice(forkChoice *engineapi.ForkChoiceMessage, headHeight uint64) {
```
Comment (enriavil1) [inline turbo/stages/headerdownload/header_algos.go:1298]: `SetUnsettledForchoice` typo

File: turbo/stages/headerdownload/header_algos.go [hunk 3]
```diff
@@ -1289,24 +1289,24 @@ func (hd *HeaderDownload) ClearPendingPayloadStatus() {
...
 	hd.lock.Lock()
 	defer hd.lock.Unlock()
-	hd.pendingHeaderHash = blockHash
-	hd.pendingHeaderHeight = blockHeight
+	hd.unsettledForkChoice = forkChoice
+	hd.unsettledHeadHeight = headHeight
 }
 
-func (hd *HeaderDownload) ClearPendingHeader() {
+func (hd *HeaderDownload) ClearUnsettledForchoice() {
```
Comment (enriavil1) [inline turbo/stages/headerdownload/header_algos.go:1305]: typo `ClearUnsettledForchoice`

File: eth/stagedsync/stage_headers.go [hunk 4]
```diff
@@ -120,34 +120,9 @@ func SpawnStageHeaders(
...
-		if !useExternalTx {
-			if err := tx.Commit(); err != nil {
-				return err
-			}
-		}
-
-		cfg.hd.ClearPendingHeader()
-
-		return nil
+	unsettledForchoice, headHeight := cfg.hd.GetUnsettledForchoice()
```
Comment (enriavil1) [inline eth/stagedsync/stage_headers.go:123]: typo `unsettledForchoice`

File: turbo/stages/headerdownload/header_algos.go (hunk 2 above)
Comment (yperbasis) [reply turbo/stages/headerdownload/header_algos.go]: Thank you! Fixed.

File: turbo/stages/headerdownload/header_algos.go (hunk 3 above)
Comment (yperbasis) [reply turbo/stages/headerdownload/header_algos.go]: Thank you! Fixed.

File: eth/stagedsync/stage_headers.go (hunk 4 above)
Comment (yperbasis) [reply eth/stagedsync/stage_headers.go]: Thank you! Fixed.

**Review by Giulio2002 — APPROVED** [review]

**Pull request merged**